### ISSUE TITLE
CPU goes high and device gets hot after 2-3 minutes

### ISSUE THREAD

**Issue by dinhthiet2702** [yoheimuta/RxMusicPlayer#65]
Hi, cpu status is high and device gets hotter after 2-3 minutes, is there any way to reduce CPU when listening,

I found there is a library that uses only 1-2% cpu, can you combine to include your perfect library?
https://github.com/tanhakabir/SwiftAudioPlayer

Thanks for the reply
<img width="364" alt="Screen Shot 2022-09-17 at 12 49 04 am" src="https://user-images.githubusercontent.com/58399579/190702003-f2390a55-a5ef-492e-8034-17cf05e79c79.png">

**Comment by rastaman111** [comment]
Hi.  how do you use this library?  I've been using it for 3 years and have no such problems.  maybe you're picking something up?

**Comment by dinhthiet2702** [comment]
> Hi. how do you use this library? I've been using it for 3 years and have no such problems. maybe you're picking something up?

hi, is your device hot?  I use it as instructed in the example.  on a real device when I exit to the background and listen for about 2-3 minutes.  app will be terminal and my device shows signs of temperature rise

**Comment by rastaman111** [comment]
very strange, maybe it is related to ios 16?
@yoheimuta Help

**Comment by dinhthiet2702** [comment]
> very strange, maybe it is related to ios 16?
> @yoheimuta Help

yes, i just upgraded ios 16 yesterday 😬😬😬.  have you been up yet?

**Comment by yoheimuta** [comment]
Thank you for letting me know. 
I don't know if it's related to this library in general.

I haven't upgraded ios 16. 
I'll take a look, but it takes time to set up ios 16.

**Comment by rastaman111** [comment]
> > very strange, maybe it is related to ios 16?
> 
> > @yoheimuta Help
> 
> 
> 
> yes, i just upgraded ios 16 yesterday 😬😬😬.  have you been up yet?

Are you testing on a simulator or on a device?  if on the simulator run multiple variants of iOS

**Comment by dinhthiet2702** [comment]
> > > very strange, maybe it is related to ios 16?
> > 
> > 
> > > @yoheimuta Help
> > 
> > 
> > yes, i just upgraded ios 16 yesterday 😬😬😬.  have you been up yet?
> 
> Are you testing on a simulator or on a device? if on the simulator run multiple variants of iOS

the above image is me running on emulator, but on real device i tested on ios 16. Previously my colleague also talked about this problem on ios 15

**Comment by yoheimuta** [comment]
> I haven't upgraded ios 16.

I've just upgraded ios 16 and run tracks repeatedly for several minutes.
I couldn't reproduce your issue. The debug navigator shows no sign of any resource spike.

My setting:

- Device (iOS16/iPhone 12 Pro)
- App ([Example](https://github.com/yoheimuta/RxMusicPlayer/tree/master/Example) and [ExampleSwiftUI](https://github.com/yoheimuta/RxMusicPlayer/tree/master/ExampleSwiftUI))
- Xcode Version 14.0 (14A309)

@dinhthiet2702 Can you reproduce this issue on the example project this repository has maintained? 
See also https://github.com/yoheimuta/RxMusicPlayer#bug-report.
